param/parameterized.py

```diff
@@ -2807,8 +2807,8 @@ def state_pop(self):
 
     @classmethod
     @Parameters.deprecate
-    def _add_parameter(cls, param_name,param_obj):
-        return cls.param._add_parameter(param_name,param_obj)
+    def add_parameter(cls, param_name,param_obj):
+        return cls.param.add_parameter(param_name,param_obj)
```

Comment:
```suggestion
    def _add_parameter(cls, param_name,param_obj):
        return cls.param.add_parameter(param_name,param_obj)
```

This one is a deprecated method, so we should leave it with the old name until it's deleted.  But we do need to have it call the newly public method on param, which I've left it doing.